lib/saluki-config/src/lib.rs

```diff
@@ -322,37 +341,59 @@ impl ConfigurationLoader {
 
     /// Consumes the configuration loader and wraps it in a generic wrapper.
     pub async fn into_generic(self) -> Result<GenericConfiguration, ConfigurationError> {
-        let inner: Value = self.inner.extract()?;
-
-        let (refreshable_config, refreshable_values) = if self.dynamic_configuration_enabled {
-            let refresher_settings = inner.deserialize::<RefresherConfiguration>()?;
-            debug!("Dynamic configuration enabled.");
-
-            let values = Arc::new(ArcSwap::from_pointee(serde_json::Value::Null));
-            let refreshable = refresher_settings.build(values.clone()).await.context(Generic)?;
-            (Some(refreshable), Some(values))
+        let refreshable_values = if self.dynamic_configuration_enabled {
```

Comment:
Dropping this field, as mentioned above, would let us get rid of checks like these and instead just directly reference `dynamic_values`.